index.d.ts

```diff
@@ -24,65 +24,80 @@ declare namespace open {
 		readonly background?: boolean;
 
 		/**
-		Specify the app to open the `target` with, or an array with the app and app arguments.
+		Specify the `name` of the app to open the `target` with and optionally, app `arguments`. `app` can be an array of apps to try to open and `name` can be an array of app names to try. If each app fails, the last error will be thrown.
 
-		The app name is platform dependent. Don't hard code it in reusable modules. For example, Chrome is `google chrome` on macOS, `google-chrome` on Linux and `chrome` on Windows.
+		The app name is platform dependent. Don't hard code it in reusable modules. For example, Chrome is `google chrome` on macOS, `google-chrome` on Linux and `chrome` on Windows. If possible, use [`open.apps`](#openapps) which auto-detects the correct binary to use.
 
 		You may also pass in the app's full path. For example on WSL, this can be `/mnt/c/Program Files (x86)/Google/Chrome/Application/chrome.exe` for the Windows installation of Chrome.
 		*/
-		readonly app?: string | readonly string[];
-
-		/**
-		__deprecated__
-
-		This option will be removed in the next major release.
-		*/
-		readonly url?: boolean;
+		readonly app?: App | readonly App[];
 
 		/**
 		Allow the opened app to exit with nonzero exit code when the `wait` option is `true`.
-		
+
 		We do not recommend setting this option. The convention for success is exit code zero.
 
 		@default false
 		*/
 		readonly allowNonzeroExitCode?: boolean;
 	}
+
+	type AppName =
+		| 'chrome'
+		| 'firefox';
+
+	type App = {name: string | readonly string[]; arguments?: readonly string[]};
 }
 
-/**
-Open stuff like URLs, files, executables. Cross-platform.
+declare const open: {
+	/**
+	An object containing auto-detected binary names for common apps. Useful to work around cross-platform differences.
+
+	@example
+	```
+	import open = require('open');
+
+	await open('https://google.com', {
+		app: {
+			name: open.apps.chrome
+		}
+	});
+	```
+	*/
+	apps: Record<open.AppName, string | string[]>;
```

Comment:
@Richienb I noticed after merging that this is not correct.

Reply:
@sindresorhus Ok, I'll fix

Reply:
As it turns out, it **is** correct.